packages/terra-form-select/src/_Frame.jsx

```diff
@@ -329,6 +329,12 @@ class Frame extends React.Component {
     });
 
     if (this.props.onSelect) {
+      if (!Util.allowsMultipleSelections(this.props)) {
```

Comment:
We should avoid calling setState twice within the same function / render lifecycle and pull this logic out of the logic block for `this.props.onSelect` which is an optional prop and shouldn't influence the outcome of isAbove.

isAbove should always be false if the drop down is closed, and not change if the dropdown remains open.

Here's an example of something that might work.

```
  handleSelect(value, option) {
    const { isAbove } = this.state;
    const isOpen = Util.allowsMultipleSelections(this.props);

    this.setState({
      searchValue: '',
      hasSearchChanged: false,
      isOpen,
      isAbove: isOpen ? isAbove : false,
    });

    if (this.props.onSelect) {
      this.props.onSelect(value, option);
    }
  }
```

Reply:
Definitely works better. I'll get this at some point, probably. :) - [41280d9](https://github.com/cerner/terra-core/pull/2170/commits/41280d9331e938f7d2b94d76bf58c9b495142fb6)